### ISSUE TITLE
Add method to separate tasks in system to allow image builds

### ISSUE THREAD

**Issue by elauqsap** [elastic/ansible-elastic-cloud-enterprise#56]
I would like to use an image building tool like Packer with your Ansible role. However, there are a few issues with the role today that make it difficult to do so. Specifically the reboot in [system/main.yml](https://github.com/elastic/ansible-elastic-cloud-enterprise/blob/master/tasks/system/main.yml) causes timeout issues and certain tasks in this file cannot be done at image build time. A possible solution would be to add two more tags for `system` (i.e `reboot` and `setup`) so that these tasks can be included during image builds but excluded during deployments.  
```yml
---
- name: Include OS specific vars
  include_vars: "{{ item }}"
  with_first_found:
  - os_{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml
  - unsupported.yml

- name: Check that OS is supported
  fail:
    msg: "ERROR: OS {{ ansible_distribution }} {{ ansible_distribution_major_version}} is not supported!"
  when: unsupported_version is defined and unsupported_version

- name: Assert docker version is supported
  assert:
    that: "docker_version in docker_version_map.keys()"
    msg: "Docker version must be one of {{ docker_version_map.keys() }}"

- name: execute os specific tasks
  include_tasks: "{{ ansible_distribution }}-{{ ansible_distribution_major_version}}/main.yml"
  tags: [setup]

- include_tasks: general/make_user.yml
  tags: [setup]
- include_tasks: general/set_limits.yml
  tags: [setup]
- include_tasks: general/setup_xfs.yml
  tags: [setup_filesystem, destructive]
  when: ansible_lvm['vgs']['lxc'] is not defined or force_xfc == true
- include_tasks: general/update_grub_docker.yml
  tags: [setup_filesystem, destructive]
- include_tasks: general/configure_docker.yml
  tags: [install_docker, destructive]
- include_tasks: general/sysctl_scripts.yml
  tags: [setup]
- include_tasks: general/kernel_modules.yml
  tags: [setup]

- name: Reboot the machine with all defaults
  shell: sleep 2 && shutdown -r now "Reboot for changes to take effect"
  async: 1
  poll: 0
  ignore_errors: true
  tags: [reboot]

- name: Wait for the reboot to complete
  wait_for_connection:
    connect_timeout: 20
    sleep: 5
    delay: 5
    timeout: 600
  tags: [reboot]

- include_tasks: general/setup_mount_permissions.yml
  tags: [setup_filesystem]
```

**Comment by cstegm** [comment]
Hey Not using ansibles reboot Module? https://docs.ansible.com/ansible/latest/modules/reboot_module.html

**Comment by cstegm** [comment]
Adding tags ist a good idea.

**Comment by elauqsap** [comment]
No the project maintainer is not using the reboot module. The proposal I provided is just a tag modified version of [system/main.yml](https://github.com/elastic/ansible-elastic-cloud-enterprise/blob/master/tasks/system/main.yml). Anyways, it would be nice if the project split tasks that can be done pre-deployment like docker install, setting limits, enabling kernel modules, etc. From tasks that can be done at deployment like file system and ECE roles. This would save anywhere from 5-10 minutes per host during deployment or upgrades.

**Comment by vaubarth** [comment]
This makes sense, might get possibly adressed in #55 if not we can adress it in a PR afterwards

**Comment by vaubarth** [comment]
This should be fixed by #55
